src/Relogify/AddResult.fs

```diff
@@ -1,48 +1,125 @@
 module Relogify.AddResult
 
+open System
 open Fabulous
 open Fabulous.XamarinForms
 open Xamarin.Forms
+open Relogify.Graphql
 
-type Model =
-    { OwnName: string
-      OpponentName: string
-      OwnPoints: int
+type ResultModel =
+    { OwnPoints: int
       OpponentPoints: int
-      ExtraTime: bool
-      CanAddResult: bool }
+      ExtraTime: bool }
+
+type State =
+    | EditingResult
+    | AddingResult
+    | ErrorAddingResult of string
+
+type Model =
+    { resultModel: ResultModel
+      state: State }
 
 let initModel =
-    { OwnName = "Player1"
-      OpponentName = "Player2"
-      OwnPoints = 0
-      OpponentPoints = 0
-      CanAddResult = false
-      ExtraTime = false }
+    { resultModel =
+        { OwnPoints = 0
+          OpponentPoints = 0
+          ExtraTime = false }
+      state = EditingResult }
 
 type Msg =
     | SetOwnPoints of int
     | SetOpponentPoints of int
     | ToggleExtraTime
+    | AddResultInitiated
+    | ResultAddedSuccess
```

Comment:
I think it is confusing, there is a message `ResultAddedSuccess`, but the `update` function doesn't care about it. So my first reaction would be to remove it, but then it will appear that the parent is using it.
It feels like the parent is depending on the internal types of the child-view, which doesn't seem correct.
I would probably create an `OutMsg` just to be explicit about the purpose of the message and that the parent might listen to it if he wants to.

Reply:
This is not a show-stopper, just a thought/suggestion 😄 Doesn't prevent from merging the PR.

Reply:
Sure, this makes sense. I made a quick attempt and will probably need help with the type signatures. Merging this for now.